code42_on_poll_connector.py

```diff
@@ -0,0 +1,287 @@
+    Actor,
+    EventTimestamp,
+    EventType,
+    FileCategory,
+)
+
+from code42_util import get_thirty_days_ago, build_alerts_query
+
+"""The contents of this module that related to mapping alert observations to file events borrows heavily from the
+Code42 Cortex XSOAR integration.
+"""
+
+
+def get_file_category_value(key):
+    # Meant to handle all possible cases
+    key = key.lower().replace("-", "").replace("_", "")
+    category_map = {
+        "sourcecode": FileCategory.SOURCE_CODE,
+        "audio": FileCategory.AUDIO,
+        "executable": FileCategory.EXECUTABLE,
+        "document": FileCategory.DOCUMENT,
+        "image": FileCategory.IMAGE,
+        "pdf": FileCategory.PDF,
+        "presentation": FileCategory.PRESENTATION,
+        "script": FileCategory.SCRIPT,
+        "spreadsheet": FileCategory.SPREADSHEET,
+        "video": FileCategory.VIDEO,
+        "virtualdiskimage": FileCategory.VIRTUAL_DISK_IMAGE,
+        "archive": FileCategory.ZIP,
+    }
+    return category_map.get(key, "UNCATEGORIZED")
+
+
+class Code42OnPollConnector:
+    def __init__(self, connector, client, state):
+        super(Code42OnPollConnector, self).__init__()
+        self._connector = connector
+        self._client = client
+        self._state = state or {}
+
+    def handle_on_poll(self, param, action_result):
+        last_time = self._state.get("last_time")
+        param = _adjust_date_parameters(last_time, param)
+        query = build_alerts_query(param["start_date"], param.get("end_date"))
+        response = self._client.alerts.search(query)
+
+        for alert in response["alerts"]:
+            details = self._get_alert_details(alert["id"])
+            container_id = self._init_container(details)
+            observations = details.get("observations", [])
+            for observation in observations:
+                file_events = self._get_file_events(observation, details)
+                self._save_artifacts_from_file_events(
+                    container_id, details, file_events
+                )
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _get_alert_details(self, alert_id):
+        return self._client.alerts.get_details(alert_id).data["alerts"][0]
+
+    def _init_container(self, alert_details):
+        container_label = self._get_container_label()
+        container_json = _create_container(alert_details, container_label)
+        _, _, container_id = self._connector.save_container(container_json)
+        return container_id
+
+    def _get_container_label(self):
+        return self._connector.get_config().get("ingest", {}).get("container_label")
+
+    def _get_file_events(self, observation, alert_details):
+        query = _get_file_event_query(observation, alert_details)
+        response = self._client.securitydata.search_file_events(query)
+        file_events = response.data.get("fileEvents", [])
+        return [_stringify_lists_if_needed(evt) for evt in file_events]
```

Comment:
Could we call this `e` or `event` rather than `evt`? Seeing `evt` makes my brain think this represents an instance of an acronym that doesn't exist